core/src/main/scala/org/apache/spark/ui/WebUI.scala

```diff
@@ -140,7 +140,8 @@ private[spark] abstract class WebUI(
   def initialize(): Unit
 
   def initServer(): ServerInfo = {
-    val hostName = Utils.localHostNameForURI()
+    val hostName = Option(conf.getenv("SPARK_LOCAL_IP"))
+        .getOrElse(if (Utils.preferIPv6) "[::]" else "0.0.0.0")
```

Comment:
This is identical with the previous behavior for IPv4 behavior and extended for IPv6.